### PR TITLE
add docker_login_with_iid

### PR DESCRIPTION
**Pull request by tuxlife**
https://github.com/zalando-stups/taupage/issues/381

### CODE REVIEW

**Comment by Raffo** [comment]
👍

**Comment by hjacobs** [comment]
@tuxlife can you add tests?

**Comment by hjacobs** [comment]
Please check the interface to comply with https://github.com/zalando-stups/pierone/pull/133

File: pierone/api.py [hunk 1]
```diff
@@ -79,6 +79,30 @@ def docker_login_with_token(url, access_token):
...
+    try:
+        with open(path) as fd:
+            dockercfg = json.load(fd)
+    except:
+        dockercfg = {}
+    document_req = request('http://169.254.169.254', '/latest/dynamic/instance-identity/document')
+    document = codecs.encode(document_req.content, 'base64').strip().decode('utf-8')
+    pkcs7_req = request('http://169.254.169.254', '/latest/dynamic/instance-identity/pkcs7')
+    pkcs7 = codecs.encode(pkcs7_req.content, 'base64').strip().decode('utf-8')
+    basic_auth = codecs.encode('{}:{}'.format(document, pkcs7).encode('utf-8'), 'base64').strip().decode('utf-8')
```
Comment (hjacobs) [inline pierone/api.py]: username is supposed to be `instance-identity-document` according to https://github.com/zalando-stups/pierone/pull/133

**Comment by tuxlife** [comment]
@hjacobs Please review

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**